File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/model/ChatComparator.java
package com.jesusc24.xroadsthroughthecastle.data.model;

import java.util.Comparator;

public class ChatComparator implements Comparator {

    //Ordena por si es privado o público el chat
    @Override
    public int compare(Object o1, Object o2) {
        return ((Chat)o2).getTipo().compareTo(((Chat)o2).getTipo());
    }
}

File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/model/ChatComparator.kt
package com.jesusc24.xroadsthroughthecastle.data.model

import java.lang.Boolean
import kotlin.Comparator
import kotlin.Int

class ChatComparator : Comparator<Chat> {

    //Ordena por si esta en favorito o no
    override fun compare(p0: Chat, p1: Chat): Int {
        return Boolean.compare(p1.favorito, p0.favorito)
    }
}